packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -5303,6 +5326,15 @@ class InheritedElement extends ProxyElement {
 
   final Map<Element, Object?> _dependents = HashMap<Element, Object?>();
 
+  /// Whether dependent widgets should unsubscribe to this [InheritedElement]
```

Comment:
The flag is meant to be used for build tacking. Using it inside didChangeDependencies doesn't make sense.
We could have an assert to prevent such usage.

Reply:
I've added an assert that only allows adding a dependency on such InheritedElement inside `build`.
Using it in initState/didChangeDependencies/... will throw

I added an exception case for LayoutBuilders too

Reply:
> The flag is meant to be used for build tacking. Using it inside didChangeDependencies doesn't make sense.

This should be documented a part of the doc comment.